facilitator/src/aws_credentials.rs

```diff
@@ -683,28 +709,27 @@ fn get_caller_identity_token_at_time(
     let mut headers = HeaderMap::new();
     headers.insert(
         HeaderName::from_static(HOST_HEADER),
-        host.parse()
-            .context("failed to parse host as header value")?,
+        host.parse().map_err(|_| StsError::BadHeaderHost)?,
     );
     headers.insert(
         HeaderName::from_static(AMAZON_DATE_HEADER),
         request_time
             .format(LONG_DATETIME)
             .to_string()
             .parse()
-            .context("failed to parse formatted date as header value")?,
+            .map_err(|_| StsError::BadHeaderDate)?,
```

Comment:
Currently, `InvalidHeaderValue` from the http crate unconditionally prints "failed to parse header value" on these errors. I think I'll add it to the enum variants just so it's not different anymore, and in case of future improvements from http.